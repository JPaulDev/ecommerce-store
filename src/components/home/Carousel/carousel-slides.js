import xsLogo from '../../../../public/images/carousel/3xs-logo.webp';
import nvidiaBackground from '../../../../public/images/carousel/nvidia-background.webp';
import intelBackground from '../../../../public/images/carousel/intel-background.webp';
import amdBackground from '../../../../public/images/carousel/amd-background.webp';
import monthBackground from '../../../../public/images/carousel/month-background.webp';
import nvidiaPc from '../../../../public/images/carousel/nvidia-pc.webp';
import intelPc from '../../../../public/images/carousel/intel-pc.webp';
import amdPc from '../../../../public/images/carousel/amd-pc.webp';
import monthPc from '../../../../public/images/carousel/month-pc.webp';
import nvidiaLogo from '../../../../public/images/carousel/nvidia-logo.webp';
import intelLogo from '../../../../public/images/carousel/intel-logo.webp';
import amdLogo from '../../../../public/images/carousel/amd-logo.webp';
import monthLogos from '../../../../public/images/carousel/month-logos.webp';

const slides = [
  {
    topNavText: '30 SERIES 3XS PCs',
    background: nvidiaBackground,
    centerImage: nvidiaPc,
    centerImageAlt: 'A PC with rainbow coloured LED case fans on the front.',
    leftLogo: xsLogo,
    leftLogoAlt: '3XS Systems.',
    leftHeading: 'RTX™ 30 Series PCs',
    leftText: `Custom built PCs built with NVIDIA's 2nd gen Ampere architecture,
      designed to deliver ultimate performance for gamers and creators.`,
    rightHeading: 'Powered by',
    rightLogo: nvidiaLogo,
    rightLogoAlt: 'Nvidia GeForce RTX.',
    rightText: [
      'Premium built to order systems',
      'Extensively tested',
      '7 day evening and weekend support',
      'Premium 3 year warranty',
      'Best in-class components',
      'Award-winning systems',
    ],
  },
  {
    topNavText: 'INTEL 3XS PCs',
    background: intelBackground,
    centerImage: intelPc,
    centerImageAlt: 'A PC with white LED fans and a custom water cooling loop.',
    leftLogo: xsLogo,
    leftLogoAlt: '3XS Systems.',
    leftHeading: 'Upgrade to 12\u1D57\u02B0 Gen Intel',
    leftText: `Level-up your PC for the maximum performance boost,
      powered by 12th Gen Intel® Core™ CPUs, with support for PCIe 5.0 graphics
      and DDR5 memory.`,
    rightHeading: 'Powered by',
    rightLogo: intelLogo,
    rightLogoAlt: 'Intel.',
    textColor: 'white',
    rightText: [
      'Premium built to order systems',
      'Extensively tested',
      '7 day evening and weekend support',
      'Premium 3 year warranty',
      'Best in-class components',
      'Award-winning systems',
    ],
  },
  {
    topNavText: 'AMD POWERED GAMING',
    background: amdBackground,
    centerImage: amdPc,
    centerImageAlt: 'A red themed PC with a custom water cooling loop.',
    leftLogo: xsLogo,
    leftLogoAlt: '3XS Systems.',
    leftHeading: 'AMD Powered Gaming',
    leftText: `3XS Gaming PCs powered by AMD Ryzen & Radeon, designed and built
      with zero compromises.`,
    rightHeading: 'Powered by',
    rightLogo: amdLogo,
    rightLogoAlt: 'AMD Ryzen 7000 series.',
    textColor: 'white',
    rightText: [
      'Premium built to order systems',
      'Extensively tested',
      '7 day evening and weekend support',
      'Premium 3 year warranty',
      'Best in-class components',
      'Award-winning systems',
    ],
  },
  {
    topNavText: 'PC OF THE MONTH',
    background: monthBackground,
    centerImage: monthPc,
    centerImageAlt: 'A custom built PC with minimal LED fans.',
    leftLogo: xsLogo,
    leftLogoAlt: '3XS Systems.',
    leftHeading: 'PC of the Month',
    leftText: `Our best deal this month, save £100 when you level-up your gaming
      with the 3XS Gamer PBA powered by RTX 3060 Ti.`,
    rightHeading: 'Powered by',
    rightLogo: monthLogos,
    rightLogoAlt: 'Nvidia GeForce RTX paired with an AMD or Intel processor.',
    rightText: [
      '8GB ASUS GeForce RTX 3060',
      'Intel Core i5 12400F or AMD Ryzen 5 5600X',
      'iCUE programmable RGB lighting',
      'Extensively tested',
      'Award-winning systems',
      '3 year warranty',
    ],
  },
];

export default slides;
