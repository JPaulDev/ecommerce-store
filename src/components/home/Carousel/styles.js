import styled, { css } from 'styled-components';

const navigationMixin = css`
  display: flex;
  list-style: none;
  position: absolute;
  z-index: 10;
`;

const buttonMixin = css`
  cursor: pointer;
  user-select: none;
`;

export const Section = styled.section`
  box-shadow: var(--box-shadow-lg);
  display: flex;
  overflow: hidden;
  position: relative;
  width: 100%;
`;

export const NavigationTop = styled.ul`
  background-color: rgb(83 83 83 / 80%);
  border-bottom: 1px solid var(--color-grey-7);
  justify-content: space-around;
  top: 0;
  width: 100%;
  ${navigationMixin}

  @media screen and (max-width: 714px) {
    display: none;
  }
`;

export const ButtonTop = styled.button`
  border-bottom: ${({ isActive }) =>
    isActive && '3px solid var(--color-orange-1)'};
  color: var(--color-grey-8);
  font-family: 'Uni Sans', sans-serif;
  font-size: var(--font-size-13);
  font-weight: var(--font-weight-semibold);
  padding: 16px 20px 12px;
  transform: ${({ isActive }) => isActive && 'scale(1.06)'};
  ${buttonMixin}

  &:hover {
    color: var(--color-grey-12);
  }

  && {
    color: ${({ isActive }) => isActive && 'white'};
  }
`;

export const NavigationBottom = styled.ul`
  align-items: center;
  bottom: 15px;
  left: 50%;
  transform: translateX(-50%);
  ${navigationMixin}

  li {
    display: flex;
  }

  @media screen and (min-width: 715px) {
    display: none;
  }
`;

export const ButtonBottom = styled.button`
  background: ${({ isActive }) =>
    isActive ? 'var(--color-orange-1)' : 'white'};
  border-radius: 2px;
  height: ${({ isActive }) => (isActive ? '7px' : '5px')};
  margin: 0 5px;
  transition: all 100ms ease-in-out;
  width: 33px;
  ${buttonMixin}
`;

export const Navigation = styled.ul`
  justify-content: space-between;
  padding: 0 15px;
  top: 50%;
  transform: translateY(-50%);
  width: 100%;
  ${navigationMixin}
`;

export const NavigationArrow = styled.button`
  align-items: center;
  background-color: white;
  border-radius: 50%;
  box-shadow: var(--box-shadow-xl);
  display: flex;
  height: 32px;
  justify-content: center;
  width: 32px;
  ${buttonMixin}
`;
