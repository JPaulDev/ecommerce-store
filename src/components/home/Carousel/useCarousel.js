import { useEffect, useReducer, useState } from 'react';

const initialState = {
  slideIndex: 0,
  direction: 1,
};

function reducer(state, action) {
  switch (action.type) {
    case 'PREV':
      return {
        ...state,
        isTransitioning: true,
        slideIndex:
          (state.slideIndex - 1 + action.payload.length) %
          action.payload.length,
        direction: -1,
      };
    case 'NEXT':
      return {
        ...state,
        isTransitioning: true,
        slideIndex: (state.slideIndex + 1) % action.payload.length,
        direction: 1,
      };
    case 'JUMP':
      return {
        ...state,
        isTransitioning: true,
        slideIndex: action.payload.index,
        direction: state.slideIndex < action.payload.index ? 1 : -1,
      };
    case 'DONE':
      return {
        ...state,
        isTransitioning: false,
      };
    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
}

export default function useCarousel({ length, interval, transitionTime }) {
  const [state, dispatch] = useReducer(reducer, initialState);
  const [isMinimized, setIsMinimized] = useState(false);

  const handleChangeSlide = (type, index) => {
    if (state.isTransitioning) return;

    dispatch({ type, payload: { index, length } });
  };

  // Stops the carousel whilst the browser window is minimized to prevent
  // animation playing incorrectly when the browser is maximized.
  useEffect(() => {
    const handleWindowMinimized = () => setIsMinimized(!isMinimized);

    document.addEventListener('visibilitychange', handleWindowMinimized);

    return () => {
      document.removeEventListener('visibilitychange', handleWindowMinimized);
    };
  }, [isMinimized]);

  useEffect(() => {
    if (isMinimized) return undefined;

    const transitionTimeout = setTimeout(() => {
      dispatch({ type: 'DONE' });
    }, transitionTime);

    const nextSlideTimeout = setTimeout(() => {
      handleChangeSlide('NEXT');
    }, interval);

    return () => {
      clearTimeout(nextSlideTimeout);
      clearTimeout(transitionTimeout);
    };
  });

  return [state.slideIndex, state.direction, handleChangeSlide];
}
