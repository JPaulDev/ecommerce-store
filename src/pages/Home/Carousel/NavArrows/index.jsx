import MaterialIcon from '../../../../components/MaterialIcon';
import * as Styled from './styles';

export default function NavArrows({ onChangeSlide }) {
  return (
    <>
      <Styled.PrevSlideBtn
        type="button"
        title="Previous"
        aria-label="Previous Slide"
        onClick={() => onChangeSlide('PREV')}
      >
        <MaterialIcon iconSize="32px">chevron_left</MaterialIcon>
      </Styled.PrevSlideBtn>
      <Styled.NextSlideBtn
        type="button"
        title="Next"
        aria-label="Next Slide"
        onClick={() => onChangeSlide('NEXT')}
      >
        <MaterialIcon iconSize="32px">chevron_right</MaterialIcon>
      </Styled.NextSlideBtn>
    </>
  );
}
