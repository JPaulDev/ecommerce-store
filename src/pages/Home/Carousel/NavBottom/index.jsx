import * as Styled from './styles';

export default function NavBottom({ length, slideIndex, onChangeSlide }) {
  return (
    <Styled.Container>
      <Styled.List>
        {Array.from({ length }).map((_, index) => (
          <li key={index}>
            <Styled.Button
              type="button"
              isActive={index === slideIndex}
              aria-label={`Slide ${index + 1} of ${length}`}
              onClick={() => onChangeSlide('JUMP', index)}
            />
          </li>
        ))}
      </Styled.List>
    </Styled.Container>
  );
}
