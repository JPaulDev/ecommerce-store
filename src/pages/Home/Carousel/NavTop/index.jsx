import * as Styled from './styles';

export default function NavTop({ slides, slideIndex, onChangeSlide }) {
  return (
    <Styled.Container data-testid="nav-top">
      <Styled.List>
        {slides.map((slide, index) => (
          <li key={slide.topNavText}>
            <Styled.Button
              type="button"
              isActive={index === slideIndex}
              onClick={() => onChangeSlide('JUMP', index)}
            >
              {slide.topNavText}
            </Styled.Button>
          </li>
        ))}
      </Styled.List>
    </Styled.Container>
  );
}
