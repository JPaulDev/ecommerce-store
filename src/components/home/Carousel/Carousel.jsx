import { motion, AnimatePresence } from 'framer-motion';
import { ChevronLeft, ChevronRight } from '../../icons';
import slides from './carousel-slides';
import useCarousel from './useCarousel';
import Slide from './Slide';
import * as Styled from './styles';

// Direction 1 moves slides in from the right, -1 moves slides in from the left

const MotionSlide = motion(Slide);
const motionProps = {
  variants: {
    enter: (direction) => ({
      x: direction > 0 ? '100%' : '-100%',
    }),
    center: {
      x: 0,
    },
    exit: (direction) => ({
      x: direction < 0 ? '100%' : '-100%',
    }),
  },
  transition: {
    ease: 'easeInOut',
    duration: 1,
  },
};

export default function Carousel() {
  const [slideIndex, direction, handleChangeSlide] = useCarousel({
    length: slides.length,
    interval: 9000,
    transitionTime: 1000,
  });

  return (
    <Styled.Section>
      <Styled.NavigationTop>
        {slides.map((slide, index) => (
          <li key={slide.topNavText}>
            <Styled.ButtonTop
              type="button"
              isActive={index === slideIndex}
              onClick={() => handleChangeSlide('JUMP', index)}
            >
              {slide.topNavText}
            </Styled.ButtonTop>
          </li>
        ))}
      </Styled.NavigationTop>
      <Styled.Navigation>
        <li>
          <Styled.NavigationArrow
            type="button"
            title="Previous"
            aria-label="Previous Slide"
            onClick={() => handleChangeSlide('PREV')}
          >
            <ChevronLeft width={14} height={19} />
          </Styled.NavigationArrow>
        </li>
        <li>
          <Styled.NavigationArrow
            type="button"
            title="Next"
            aria-label="Next Slide"
            onClick={() => handleChangeSlide('NEXT')}
          >
            <ChevronRight width={14} height={19} />
          </Styled.NavigationArrow>
        </li>
      </Styled.Navigation>
      <AnimatePresence initial={false} custom={direction}>
        <MotionSlide
          key={slideIndex}
          custom={direction}
          initial="enter"
          animate="center"
          exit="exit"
          slideIndex={slideIndex}
          slide={slides[slideIndex]}
          {...motionProps}
        />
      </AnimatePresence>
      <Styled.NavigationBottom>
        {Array.from({ length: slides.length }).map((_, index) => (
          <li key={index}>
            <Styled.ButtonBottom
              type="button"
              isActive={index === slideIndex}
              aria-label={`Slide ${index + 1} of ${slides.length}`}
              onClick={() => handleChangeSlide('JUMP', index)}
            />
          </li>
        ))}
      </Styled.NavigationBottom>
    </Styled.Section>
  );
}
