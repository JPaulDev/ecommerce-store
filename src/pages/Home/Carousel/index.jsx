import { motion, AnimatePresence } from 'framer-motion';
import useMediaQuery from '../../../hooks/useMediaQuery';
import useCarousel from './useCarousel';
import slides from './carousel-slides';
import NavTop from './NavTop';
import NavBottom from './NavBottom';
import NavArrows from './NavArrows';
import Slide from './Slide';
import * as Styled from './styles';

const MotionSlide = motion(Slide);

// Direction 1 moves slides to left, -1 moves slides to the right
const variants = {
  enter: (direction) => ({
    translateX: direction > 0 ? '100%' : '-100%',
  }),
  center: {
    translateX: 0,
  },
  exit: (direction) => ({
    translateX: direction < 0 ? '100%' : '-100%',
  }),
};

export default function Carousel() {
  const isMatch = useMediaQuery('(min-width: 700px)');
  const [slideIndex, direction, handleChangeSlide] = useCarousel({
    length: slides.length,
    interval: 8000,
    transitionTime: 1000,
  });

  return (
    <Styled.Section>
      {isMatch ? (
        <NavTop
          slides={slides}
          slideIndex={slideIndex}
          onChangeSlide={handleChangeSlide}
        />
      ) : (
        <NavBottom
          length={slides.length}
          slideIndex={slideIndex}
          onChangeSlide={handleChangeSlide}
        />
      )}
      <NavArrows onChangeSlide={handleChangeSlide} />
      <AnimatePresence initial={false} custom={direction}>
        <MotionSlide
          key={slideIndex}
          custom={direction}
          variants={variants}
          initial="enter"
          animate="center"
          exit="exit"
          transition={{ ease: 'easeInOut', duration: 1 }}
          slide={slides[slideIndex]}
        />
      </AnimatePresence>
    </Styled.Section>
  );
}
