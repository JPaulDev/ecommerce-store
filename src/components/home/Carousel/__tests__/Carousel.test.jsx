import {
  render,
  screen,
  act,
  waitForElementToBeRemoved,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Carousel from '../index';

// Prevents errors caused by next image component.
jest.mock('next/future/image', () => 'img');

jest.mock('../carousel-slides', () => [
  {
    topNavText: 'slide-1',
    rightText: [],
  },
  {
    topNavText: 'slide-2',
    rightText: [],
  },
  {
    topNavText: 'slide-3',
    rightText: [],
  },
]);

function setup(component) {
  return {
    user: userEvent.setup(),
    ...render(component),
  };
}

describe('Carousel component', () => {
  it('renders a carousel and all of its navigation buttons', () => {
    const tree = renderer.create(<Carousel />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Carousel component side navigation arrows', () => {
  it('navigates to the next slide on clicking the button', async () => {
    const { user } = setup(<Carousel />);

    expect(screen.queryByTestId('slide-2')).not.toBeInTheDocument();

    await user.click(screen.getByRole('button', { name: /next slide/i }));

    expect(screen.getByTestId('slide-2')).toBeInTheDocument();
  });

  it('navigates to the last slide on clicking the previous button', async () => {
    const { user } = setup(<Carousel />);

    expect(screen.queryByTestId('slide-3')).not.toBeInTheDocument();

    await user.click(screen.getByRole('button', { name: /previous slide/i }));

    expect(screen.getByTestId('slide-3')).toBeInTheDocument();
  });

  it('goes back to the first slide on clicking the next button', async () => {
    const { user } = setup(<Carousel />);

    await user.click(screen.getByRole('button', { name: /previous slide/i }));

    await waitForElementToBeRemoved(screen.queryByTestId('slide-1'), {
      timeout: 2500,
    });

    await user.click(screen.getByRole('button', { name: /next slide/i }));

    expect(screen.getByTestId('slide-1')).toBeInTheDocument();
  });

  it('should prevent navigating to another slide while transitioning', async () => {
    const { user } = setup(<Carousel />);

    expect(screen.queryByTestId('slide-3')).not.toBeInTheDocument();

    await user.dblClick(screen.getByRole('button', { name: /next slide/i }));

    expect(screen.queryByTestId('slide-3')).not.toBeInTheDocument();
  });
});

describe('Carousel component', () => {
  it('displays the first slide on load', () => {
    setup(<Carousel />);
    expect(screen.getByTestId('slide-1')).toBeInTheDocument();
  });

  it('should display only 1 slide', () => {
    setup(<Carousel />);

    // eslint-disable-next-line jest-dom/prefer-in-document
    expect(screen.getAllByTestId(/slide/i)).toHaveLength(1);
  });

  it('changes to the next slide after 9 seconds', () => {
    jest.useFakeTimers();
    setup(<Carousel />);
    expect(screen.queryByTestId('slide-2')).not.toBeInTheDocument();

    act(() => {
      jest.advanceTimersByTime(9000);
    });

    expect(screen.getByTestId('slide-2')).toBeInTheDocument();
    jest.useRealTimers();
  });
});

describe('Carousel component top navigation buttons', () => {
  it('changes to the correct slide on clicking the 3rd button', async () => {
    const { user } = setup(<Carousel />);

    expect(screen.queryByTestId('slide-3')).not.toBeInTheDocument();

    await user.click(screen.getByRole('button', { name: /slide-3/i }));

    expect(screen.getByTestId('slide-3')).toBeInTheDocument();
  });
});

describe('Carousel component bottom navigation buttons', () => {
  it('navigates to the correct slide on clicking the 3rd button', async () => {
    const { user } = setup(<Carousel />);

    expect(screen.queryByTestId('slide-3')).not.toBeInTheDocument();

    await user.click(screen.getByRole('button', { name: /slide 3 of 3/i }));

    expect(screen.getByTestId('slide-3')).toBeInTheDocument();
  });
});
