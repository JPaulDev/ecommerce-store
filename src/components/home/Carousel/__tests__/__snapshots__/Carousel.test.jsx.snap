// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component renders a carousel and all of its navigation buttons 1`] = `
.c6 {
  background-color: black;
  margin-right: -100%;
  min-height: 450px;
  overflow: hidden;
  padding: 10px 20px 0;
  position: relative;
  width: 100%;
}

.c7 {
  position: absolute;
  right: -383px;
  top: 0;
}

.c8 {
  color: white;
  position: relative;
  text-align: center;
}

.c9 {
  display: inline-block;
  margin-top: 10px;
}

.c9 img {
  display: block;
  height: auto;
  max-width: 100%;
}

.c10 {
  font-family: 'Uni Sans',sans-serif;
  font-size: 1.93rem;
  margin-bottom: 10px;
  margin-top: 5px;
}

.c11 {
  font-size: 0.88rem;
  line-height: 1.6;
}

.c12 {
  left: 50%;
  position: absolute;
  top: 180px;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  width: 325px;
}

.c12 img {
  display: block;
  height: auto;
  max-width: 100%;
}

.c13 {
  position: relative;
  width: 30%;
}

.c14 {
  font-family: 'Uni Sans',sans-serif;
  font-size: 1.3rem;
  font-weight: 600;
  margin-left: 10px;
  text-indent: 10px;
}

.c14 img {
  margin-left: 15px;
}

.c15 {
  font-size: 0.9rem;
  padding: 10px 0 0 18px;
}

.c15 li {
  padding: 5px 0;
}

.c0 {
  box-shadow: 0 3px 5px rgb(0 0 0 / 40%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  position: relative;
  width: 100%;
}

.c1 {
  background-color: rgb(83 83 83 / 80%);
  border-bottom: 1px solid var(--color-grey-7);
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  top: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  position: absolute;
  z-index: 10;
}

.c2 {
  background: none;
  border-bottom: 3px solid var(--color-orange-1);
  color: #b9b9b9;
  font-family: 'Uni Sans',sans-serif;
  font-size: 0.83rem;
  font-weight: 600;
  padding: 16px 20px 12px;
  -webkit-transform: scale(1.06);
  -ms-transform: scale(1.06);
  transform: scale(1.06);
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2:hover {
  color: var(--color-grey-9);
}

.c2.c2 {
  color: white;
}

.c3 {
  background: none;
  color: #b9b9b9;
  font-family: 'Uni Sans',sans-serif;
  font-size: 0.83rem;
  font-weight: 600;
  padding: 16px 20px 12px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3:hover {
  color: var(--color-grey-9);
}

.c16 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  bottom: 15px;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  position: absolute;
  z-index: 10;
}

.c16 li {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c17 {
  background: var(--color-orange-1);
  border-radius: 2px;
  height: 7px;
  margin: 0 5px;
  -webkit-transition: all 100ms ease-in-out;
  transition: all 100ms ease-in-out;
  width: 33px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c18 {
  background: white;
  border-radius: 2px;
  height: 5px;
  margin: 0 5px;
  -webkit-transition: all 100ms ease-in-out;
  transition: all 100ms ease-in-out;
  width: 33px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c4 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 15px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  position: absolute;
  z-index: 10;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
  border-radius: 50%;
  box-shadow: 1px 4px 4px rgb(0 0 0 / 50%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 32px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

@media screen and (min-width:550px) {
  .c6 {
    padding: 10px 60px 0;
  }
}

@media screen and (min-width:715px) {
  .c6 {
    padding: 90px 60px 0;
    min-height: 385px;
  }
}

@media screen and (min-width:1000px) {
  .c6 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media screen and (max-width:714px) {
  .c7::after {
    background-image: linear-gradient( to right, rgb(0 0 0 / 40%), rgb(0 0 0 / 15%) );
    content: '';
    height: 100%;
    left: 0;
    position: absolute;
    width: 100%;
  }
}

@media screen and (min-width:1000px) {
  .c7 {
    right: initial;
    left: 50%;
    -webkit-transform: translateX(-50%);
    -ms-transform: translateX(-50%);
    transform: translateX(-50%);
  }
}

@media screen and (min-width:715px) {
  .c8 {
    width: 40%;
    text-align: left;
  }
}

@media screen and (min-width:1000px) {
  .c8 {
    width: 30%;
  }
}

@media screen and (max-width:714px) {
  .c9 {
    width: 125px;
  }
}

@media screen and (min-width:715px) {
  .c10 {
    margin-top: 15px;
  }
}

@media screen and (min-width:715px) {
  .c12 {
    -webkit-transform: translateX(-10%);
    -ms-transform: translateX(-10%);
    transform: translateX(-10%);
    top: 100px;
  }
}

@media screen and (min-width:850px) {
  .c12 {
    width: 350px;
    top: 90px;
    right: 110px;
    left: initial;
    -webkit-transform: initial;
    -ms-transform: initial;
    transform: initial;
  }
}

@media screen and (min-width:1000px) {
  .c12 {
    width: 325px;
    top: 100px;
    left: 50%;
    -webkit-transform: translateX(-50%);
    -ms-transform: translateX(-50%);
    transform: translateX(-50%);
  }
}

@media screen and (min-width:1100px) {
  .c12 {
    width: 375px;
    top: 80px;
  }
}

@media screen and (max-width:999px) {
  .c13 {
    display: none;
  }
}

@media screen and (max-width:714px) {
  .c1 {
    display: none;
  }
}

@media screen and (min-width:715px) {
  .c16 {
    display: none;
  }
}

<section
  className="c0"
>
  <ul
    className="c1"
  >
    <li>
      <button
        className="c2"
        onClick={[Function]}
        type="button"
      >
        slide-1
      </button>
    </li>
    <li>
      <button
        className="c3"
        onClick={[Function]}
        type="button"
      >
        slide-2
      </button>
    </li>
    <li>
      <button
        className="c3"
        onClick={[Function]}
        type="button"
      >
        slide-3
      </button>
    </li>
  </ul>
  <ul
    className="c4"
  >
    <li>
      <button
        aria-label="Previous Slide"
        className="c5"
        onClick={[Function]}
        title="Previous"
        type="button"
      >
        <svg
          fill="currentColor"
          height={19}
          viewBox="0 0 384 512"
          width={14}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l192 192c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 278.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-192 192z"
          />
        </svg>
      </button>
    </li>
    <li>
      <button
        aria-label="Next Slide"
        className="c5"
        onClick={[Function]}
        title="Next"
        type="button"
      >
        <svg
          fill="currentColor"
          height={19}
          viewBox="0 0 384 512"
          width={14}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M342.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L274.7 256 105.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"
          />
        </svg>
      </button>
    </li>
  </ul>
  <div
    className="c6"
    data-testid="slide-1"
  >
    <div
      className="c7"
    >
      <img
        alt=""
        priority="true"
        quality={90}
      />
    </div>
    <div
      className="c8"
    >
      <div
        className="c9"
      >
        <img />
      </div>
      <h4
        className="c10"
      />
      <p
        className="c11"
      />
    </div>
    <div
      className="c12"
    >
      <img
        quality={90}
      />
    </div>
    <div
      className="c13"
    >
      <h4
        className="c14"
      >
        <img />
      </h4>
      <ul
        className="c15"
      />
    </div>
  </div>
  <ul
    className="c16"
  >
    <li>
      <button
        aria-label="Slide 1 of 3"
        className="c17"
        onClick={[Function]}
        type="button"
      />
    </li>
    <li>
      <button
        aria-label="Slide 2 of 3"
        className="c18"
        onClick={[Function]}
        type="button"
      />
    </li>
    <li>
      <button
        aria-label="Slide 3 of 3"
        className="c18"
        onClick={[Function]}
        type="button"
      />
    </li>
  </ul>
</section>
`;
